modules/RTC/ortc/RTCSessionDescription.js

```diff
@@ -0,0 +1,106 @@
+import sdpTransform from 'sdp-transform';
+
+/**
+ * RTCSessionDescription implementation.
+ */
+export default class RTCSessionDescription {
+    /**
+     * RTCSessionDescription constructor.
+     * @param {object} [descriptionInitDict]
+     * @param {string} [descriptionInitDict.type] "offer" / "answer".
+     * @param {string|object} [descriptionInitDict.sdp] SDP string or
+     * sdp-transform object.
+     */
+    constructor(descriptionInitDict) {
+        const type = descriptionInitDict.type;
+
+        switch (type) {
+        case 'offer':
+            break;
+        case 'answer':
+            break;
+        default:
+            throw new TypeError(`invalid type "${type}"`);
+        }
+
+        let sdp = descriptionInitDict.sdp;
+        let parsed;
+
+        switch (typeof sdp) {
+        case 'string':
+            try {
+                parsed = sdpTransform.parse(sdp);
+            } catch (error) {
+                throw new Error(`invalid sdp: ${error}`);
+            }
+            break;
+
+        case 'object':
+            parsed = sdp;
+            try {
+                sdp = sdpTransform.write(parsed);
+            } catch (error) {
+                throw new Error(`invalid sdp-transform object: ${error}`);
+            }
+            break;
+
+        default:
+            throw new TypeError(
+                'sdp must be a string or a sdp-transform object');
+        }
+
+        // Private attributes.
+        this._type = type;
+        this._sdp = sdp;
+        this._parsed = parsed;
+    }
+
+    /**
+     * Get type field.
+     */
+    get type() {
+        return this._type;
+    }
+
+    /**
+     * Set type field.
+     * @param {string} type
+     */
+    set type(type) {
+        this._type = type;
+    }
+
+    /**
+     * Get sdp field.
+     */
+    get sdp() {
+        return this._sdp;
+    }
+
+    /**
+     * Set sdp field.
+     * @param {string} sdp
+     */
+    set sdp(sdp) {
+        this._sdp = sdp;
+    }
+
+    /**
+     * Gets the internal sdp-transform object.
+     */
+    get parsed() {
+        return this._parsed;
+    }
+
+    /**
+     * Serialize the RTCSessionDescription into an object.
+     * @return {RTCSessionDescriptionInit}
+     * @private
+     */
+    toRTCSessionDescriptionInit() {
```

Comment:
You won't find such a `RTCSessionDescription.toJSON()` method in any version of the webrtc draft.

Anyhow and again: a JSON "object" is not a JS object, but a JS string. And that's not what we need here.

Said that, it's true that, at least Chrome, implements such an annoying  `RTCSessionDescription.toJSON()` method that, incredibly, returns a JS object rather than a string. So ok, let's do such a wrong hack.

Done.